### PR TITLE
Clarify `StoreNewHeads` logic

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

`StoreNewHeads` bundles a lot of operations together, using flags to turn some on or off. This PR breaks the method into separate methods with clear purposes (e.g. update L1 head block, store a new L1 block -> L2 rollup mapping, etc.)

### What changes were made as part of this PR:

- Provide a high level list of the changes made
- List key areas for the reviewer 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/enclave/db/interfaces.go [hunk 1]
```diff
@@ -54,8 +56,10 @@ type HeadsAfterL1BlockStorage interface {
 	FetchLogs(blockHash common.L1RootHash) ([]*types.Log, error)
 	// FetchL2Head returns the current L2 chain head.
 	FetchL2Head() (*common.L2RootHash, error)
-	// StoreNewHeads saves the new chain heads alongside its rollup, receipts and logs.
-	StoreNewHeads(l1Head common.L1RootHash, rollup *core.Rollup, receipts []*types.Receipt, isNewRollup bool, isNewL1Block bool) error
```
Comment (joeldudley) [inline go/enclave/db/interfaces.go:58]: This method has been split into three:

* `StoreRollup`, to store the new rollup
* `UpdateL2HeadForL1Block`, to update the rollup associated with a given L1 block
* `UpdateL1Head`, to update the current L1 head block

File: go/enclave/db/rawdb/accessors_chain.go [hunk 2]
```diff
@@ -16,17 +16,12 @@ import (
 )
 
 func ReadRollup(db ethdb.KeyValueReader, hash gethcommon.Hash) (*core.Rollup, error) {
-	height, err := ReadHeaderNumber(db, hash)
```
Comment (joeldudley) [inline go/enclave/db/rawdb/accessors_chain.go:19]: There were a bunch of methods where we were keying stuff in the DB by hash + number. This wasn't needed - just using the hash was sufficient.

File: go/enclave/db/storage.go [hunk 3]
```diff
@@ -43,19 +43,13 @@ func NewStorage(backingDB ethdb.Database, chainConfig *params.ChainConfig, logge
 	}
 }
 
-func (s *storageImpl) StoreGenesisRollup(rol *core.Rollup) error {
+func (s *storageImpl) StoreGenesisRollupHash(rollupHash common.L2RootHash) error {
```
Comment (joeldudley) [inline go/enclave/db/storage.go:46]: This method only updates the genesis hash now, with storing the rollup performed by another method (`StoreRollup`).

File: go/enclave/db/storage.go [hunk 4]
```diff
@@ -213,25 +207,19 @@ func (s *storageImpl) FetchLogs(blockHash common.L1RootHash) ([]*types.Log, erro
 	return logs, nil
 }
 
-// TODO - #718 - This method has behaviour that's too dependent on various flags. Decompose.
-func (s *storageImpl) StoreNewHeads(l1Head common.L1RootHash, rollup *core.Rollup, receipts []*types.Receipt, isNewRollup bool, isNewL1Block bool) error {
+func (s *storageImpl) UpdateL2HeadForL1Block(l1Head common.L1RootHash, l2Head *core.Rollup, receipts []*types.Receipt) error {
```
Comment (joeldudley) [inline go/enclave/db/storage.go:210]: This method now only makes changes related to modifying which rollup a given L1 block is associated with.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**